charm_tracing charmlib does not correctly wrap staticmethods in instrumented code

### Bug Description

The charm_tracing charm lib [has a fixme comment](https://github.com/canonical/tempo-k8s-operator/blob/main/lib/charms/tempo_k8s/v1/charm_tracing.py#L543-L546) which manifested and was observed in mysql-router-k8s-operator's unit tests (see logs below for error trace).

### To Reproduce

1. git clone git@github.com:canonical/mysql-router-k8s-operator.git
2. git checkout feature/tempo_tracing
3. virtualenv venv
4. source venv/bin/activate
5. pip install poetry tox
6. tox -e unit

### Environment

tempo_k8s charm_tracing lib: v1.6

### Relevant log output

```shell
tests/unit/test_workload.py:188: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  

args = ('# File automatically generated during MySQL Router bootstrap\n[DEFAULT]\nname=system\nkeyring_path=/var/lib/mysqlrou...e\n\n[rest_routing]\nrequire_realm=default_auth_realm\n\n[rest_metadata_cache]\nrequire_realm=default_auth_realm\n\n',)                
kwargs = {}, name = 'AuthenticatedWorkload._parse_username_from_config'                                  

    @functools.wraps(callable)
    def wrapped_function(*args, **kwargs):  # type: ignore                                               
        name = getattr(callable, "__qualname__", getattr(callable, "__name__", str(callable)))           
        with _span(f"{'(static) ' if static else ''}{qualifier} call: {name}"):  # type: ignore          
            if static:                              
                # fixme: do we or don't we need [1:]?                                                    
                #  The _trace_callable decorator doesn't always play nice with @staticmethods.           
                #  Sometimes it will receive 'self', sometimes it won't.                                 
                # return callable(*args, **kwargs)  # type: ignore                                       
>               return callable(*args[1:], **kwargs)  # type: ignore                                     
E               TypeError: AuthenticatedWorkload._parse_username_from_config() missing 1 required positional argument: 'config_file_text'
```


### Additional context

static method signature:

```
@staticmethod
def _parse_username_from_config(config_file_text: str) -> str:
```

--- a/lib/charms/tempo_k8s/v1/charm_tracing.py
+++ b/lib/charms/tempo_k8s/v1/charm_tracing.py
@@ -147,7 +147,7 @@ LIBAPI = 1
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
 
-LIBPATCH = 6
+LIBPATCH = 7
 
 PYDEPS = ["opentelemetry-exporter-otlp-proto-http==1.21.0"]
 
@@ -540,11 +540,13 @@ def _trace_callable(callable: _F, qualifier: str, static: bool = False) -> _F:
         name = getattr(callable, "__qualname__", getattr(callable, "__name__", str(callable)))
         with _span(f"{'(static) ' if static else ''}{qualifier} call: {name}"):  # type: ignore
             if static:
-                # fixme: do we or don't we need [1:]?
-                #  The _trace_callable decorator doesn't always play nice with @staticmethods.
-                #  Sometimes it will receive 'self', sometimes it won't.
-                # return callable(*args, **kwargs)  # type: ignore
-                return callable(*args[1:], **kwargs)  # type: ignore
+                # The _trace_callable decorator doesn't play super nice with @staticmethods.
+                # if you call MyObj().mystaticmethod we'll receive the MyObj instance as first argument,
+                # if you call MyObj.mystaticmethod we won't.
+                try:
+                    inspect.signature(callable).bind(*args, **kwargs)
+                except TypeError:
+                    return callable(*args[1:], **kwargs)  # type: ignore
             return callable(*args, **kwargs)  # type: ignore
 
     # wrapped_function.__signature__ = sig

--- a/tests/scenario/test_charm_tracing.py
+++ b/tests/scenario/test_charm_tracing.py
@@ -466,3 +466,53 @@ def test_borky_tempo_return_value(borky_return_value, caplog):
     # logger.exception in _setup_root_span_initializer
     assert "exception retrieving the tracing endpoint from" in caplog.text
     assert "proceeding with charm_tracing DISABLED." in caplog.text
+
+
+class MyCharmStaticMethods(CharmBase):
+    META = {"name": "jolene"}
+
+    def __init__(self, fw):
+        super().__init__(fw)
+        fw.observe(self.on.start, self._on_start)
+
+    def _on_start(self, _):
+        assert OtherObj()._staticmeth(1) == 2
+        assert OtherObj._staticmeth(1) == 2
+
+    @property
+    def tempo(self):
+        return "foo.bar:80"
+
+
+class OtherObj:
+    @staticmethod
+    def _staticmeth(i):
+        return 1 + i
+
+
+autoinstrument(MyCharmStaticMethods, MyCharmStaticMethods.tempo, extra_types=[OtherObj])
+
+
+def test_trace_staticmethods(caplog):
+    import opentelemetry
+
+    with patch(
+        "opentelemetry.exporter.otlp.proto.http.trace_exporter.OTLPSpanExporter.export"
+    ) as f:
+        f.return_value = opentelemetry.sdk.trace.export.SpanExportResult.SUCCESS
+        ctx = Context(MyCharmStaticMethods, meta=MyCharmStaticMethods.META)
+        ctx.run("start", State())
+
+        spans = f.call_args_list[0].args[0]
+
+        span_names = [span.name for span in spans]
+        assert span_names == [
+            "(static) method call: OtherObj._staticmeth",
+            "(static) method call: OtherObj._staticmeth",
+            "method call: MyCharmStaticMethods._on_start",
+            "event: start",
+            "charm exec",
+        ]
+
+        for span in spans:
+            assert span.resource.attributes["service.name"] == "jolene"

